app/views/api/v2/layouts/index_layout.json.erb

```diff
@@ -6,6 +6,7 @@
   "search": <%= metadata_search.to_json.html_safe %>,
 <% if params.has_key?(:include_permissions) %>
   "can_create": <%= can_create?.to_json %>,
+  "can_edit": <%= can_edit?.to_json %>,
```

Comment:
A bit late to the party, but what about a user who has a filter for the edit permission? e.g. someone who can only edit hosts in the development environment - this would return `true` (since they have `edit_hosts` permission), but in fact if they try to edit a host in the production environment they will get an error.
Unlike the create permission which can't really be filtered and is generic for the resource, this value could be different for each item in the index.

Reply:
This was added purely for bulk actions where we can't really check the resources in question one by one. The meaning is more like "is there a possibility that the user could edit at least one of them". It definitely doesn't mean "the user can edit all the returned resources"

Reply:
This seems like we should really get on doing something about perms in the frontend ... :thinking:

Reply:
And you're not late, we've not cherry-picked yet :shrug:  :)

Reply:
> This was added purely for bulk actions where we can't really check the resources in question one by one. The meaning is more like "is there a possibility that the user could edit at least one of them". It definitely doesn't mean "the user can edit all the returned resources"

What would happen if a user with a filtered edit permission tries to perform a bulk action?

Reply:
I don't think we have a systematic solution in place for this. In tasks (that's where this came from) it would perform the action only on resources for which the user has necessary permissions. This flag is only used to disable the button in case there's no chance a user could edit any of the tasks.

Reply:
Is this sufficient or should I start looking for a different solution in case this gets rolled back?

Reply:
I'd go with this for now, but we should keep in mind that we need a better solution in the future.
I've added this to the agenda for next UI meeting: https://community.theforeman.org/t/ui-ux-meeting-agenda/22460

Reply:
> Unlike the create permission which can't really be filtered and is generic for the resource, this value could be different for each item in the index.

This is not true for a while, since https://github.com/theforeman/foreman/pull/4030 even create/update permissions support granular filters

I agree we need some solution for frontend but I think this PR is appropriate since it's just about hiding/showing the button, not about allowing user to perform the action.